lib/src/main/java/uk/gov/di/ipv/core/library/domain/ConfigurationServicePublicKeySelector.java

```diff
@@ -40,13 +51,40 @@ public ConfigurationServicePublicKeySelector(ConfigurationService configurationS
             boolean forceRefresh,
             Context context)
             throws InvalidClientException {
-        try {
-            return List.of(
-                    configurationService
-                            .getClientCertificate(claimedClientID.getValue())
-                            .getPublicKey());
-        } catch (CertificateException e) {
-            throw new InvalidClientException(e.getMessage());
+
+        JWSAlgorithm algorithm = jwsHeader.getAlgorithm();
+        String clientId = claimedClientID.getValue();
+        String publicKeyMaterial = configurationService.getClientPublicKeyMaterial(clientId);
+
+        switch (algorithm.toString()) {
+            case ES256:
+                try {
```

Comment:
For a future refactor:
Something doesn't seem right about having the try catch within the case block. Maybe wrapping the switch within one try catch is safer? However, I'm unsure how you'd provide a specific error message - so ignore for now.